### PR TITLE
Add CI to build all the targets

### PR DESCRIPTION
**Pull request by SergioGasquez**
Expanded the CI to build the project for all the targets using the Dockerfile generated for devcontainer support.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -63,8 +63,11 @@ jobs:
         board: ['esp32', 'esp32s2', 'esp32s3', 'esp32c3']
         std: ['true', 'false']
     steps:
-      - name: Setup | cargo-generate
-        run: cargo install cargo-generate
+      - name: Install cargo-generate
```
Comment (ivmarkov) [inline .github/workflows/ci.yml:66]: Why this and not just `cargo install cargo-generate`?

**Review by ivmarkov — COMMENTED** [review]
I'm not sure why we are now installing cargo-generate that way?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (SergioGasquez) [reply .github/workflows/ci.yml:66]: I changed to the binary approach as it's way faster. If you think is better to do it with `cargo install` I am happy to undo this change

**Comment by SergioGasquez** [comment]
> I'm not sure why we are now installing cargo-generate that way?

It's only because it's faster, but I wouldn't mind using `cargo install` if you think is better

**Comment by SergioGasquez** [comment]
Installing `cargo-generate` from release binaries can also make the CI more "consistent" as it avoids some "external" errors on CI. Eg: [Todays CI run failed because `cargo install cargo-generate` was failing](https://github.com/esp-rs/esp-idf-template/runs/7354068033?check_suite_focus=true)

**Comment by ivmarkov** [comment]
@SergioGasquez Sorry for following up late, can you resolve the conflicts, then I'll merge?

**Comment by SergioGasquez** [comment]
> @SergioGasquez Sorry for following up late, can you resolve the conflicts, then I'll merge?

Hello @ivmarkov, do not worry about the delays! Conflicts are now resolved. Thanks for all your dedication!

**Comment by ivmarkov** [comment]
@SergioGasquez I think we need to temporarily disable the PlatformIO based build, until we figure out what is wrong there.

**Comment by SergioGasquez** [comment]
> @SergioGasquez I think we need to temporarily disable the PlatformIO based build, until we figure out what is wrong there.

I've commented those steps!

**Comment by ivmarkov** [comment]
@SergioGasquez BTW it still complains that the branch is unmergeable due to conflicts.
I think you need to rebase it on top of latest master first.

**Comment by SergioGasquez** [comment]
It now shows that it can be merged with no conflicts! :)

**Comment by ivmarkov** [comment]
It can be merged but it cannot be rebased. Anyway, ill not use rebase then.

**Pull request merged**